### ISSUE TITLE
hlx publish randomly fires an exception

### ISSUE THREAD

**Issue by kptdobe** [adobe/helix-cli#1484]
Follow up of yesterday escalations where `helix-pages` Fastly config was corrupted (see https://app.circleci.com/pipelines/github/adobe/helix-pages/1596/workflows/b7b6c100-88ef-4856-8b5a-10e2e8b9caae/jobs/5604).

While trying to manually publish, I got an error:

```bash
$ hlx publish --custom-vcl='vcl/extensions.vcl'
Publishing [==------------------------------------------------] preparing fastly transaction 0.0s{ progress: true, level: 'info', message: 'preparing fastly transaction' }
Publishing [====----------------------------------------------] preparing service config for Helix 11.3s{ progress: true, level: 'info', message: 'preparing service config for Helix' }
Publishing [=======================---------------------------] failed to set service config up for Helix 1.8s{ progress: true, level: 'info', message: 'failed to set service config up for Helix' }
error: Remote publish service failed Error: 400 - "{
  "code": "IdJTjZ3OmnzeYT9jsUEC25y0bAUAHU4f",
  "error": "There was an error processing your request."
}"
error: Error while running the Publish command: Error: Unable to setup service config
    at fetch.then.catch (/Users/acapt/.nvm/versions/node/v10.21.0/lib/node_modules/@adobe/helix-cli/src/remotepublish.cmd.js:378:13)
    at process._tickCallback (internal/process/next_tick.js:68:7) Error: Unable to setup service config
    at fetch.then.catch (/Users/acapt/.nvm/versions/node/v10.21.0/lib/node_modules/@adobe/helix-cli/src/remotepublish.cmd.js:378:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Error while running the Publish command
```

This still created a draft Fastly config and which has not been activated. https://manage.fastly.com/configure/services/1McGRQOYFuABWBHyD8D4Ux/versions/382

Re-running the command fixed the issue.

**Comment by tripodsan** [comment]
related error from the failed `hlx publish above`:

```
    "activationId": "13ecb901da8546caacb901da8576cab5",
    "cause": "d3c8c80687ec4f7988c80687ecaf7949",
    "start": 1595401774030,
    "end": 1595401774248,
    "duration": 218,
    "statusCode": 0,
    "response": {
        "status": "action developer error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": "The action did not produce a valid response and exited unexpectedly."
        }
    },
```

**Comment by stefan-guggisberg** [comment]
Fastly error:

```
error: Error while running the Publish command: FastlyError: Syntax error: Expected one of: `acl`, `callback`, `sub`, `backend`, `director`, `import`, `table`, `limiter`, `counter`, or `pragma`
at: (input Line 933 Pos 1)
set req.http.X-Root-Path = "";
```

That line is the content of the generated `strains.vcl`. I see 2 possible explanations:

1. `strains.vcl` somehow became the `main` script. 
1. `strains.vcl` somehow got included at the wrong location (i.e. not within the `sub hlx_strain` scope).

Both would trigger above error. I think it's the 1st one. 

Looking at the `helix-publish` code I see that setting up the Fastly service config consists of a number of tasks which are all executed in parallel:

https://github.com/adobe/helix-publish/blob/master/src/publish.js#L85-L111

Are these tasks really safe to be run in parallel? Does Fastly support concurrent config updates? Maybe `request-promise` somehow serialised these updates whereas with `helix-fetch` the updates are now concurrent? Just a thought. 

@trieloff @kptdobe @tripodsan WDYT?

**Comment by kptdobe** [comment]
Looking at the code, 2 tasks are `init` something.. I do not think the intent was to run in parallel!

**Comment by kptdobe** [comment]
Looking deeper at the code, the tasks seem to be independent thus you might be right with the Fastly API being confused with so many parallel requests. If you look at the Epsagon logs, we seem to be doing 26 requests in parallel... the API might not like it!

**Comment by stefan-guggisberg** [comment]
@kptdobe This should be fixed now via adobe/helix-publish#549. Can we close this issue?
